tgs/src/main/cpp/tgs/RandomForest/RandomForest.cpp

```diff
@@ -177,7 +177,7 @@ void RandomForest::trainMulticlass(const std::shared_ptr<DataFrame>& data, unsig
 
         std::vector<std::string> badFactors;
 
-        unsigned int cutOffIdx =
+        unsigned long cutOffIdx =
           topFactors.size() - (unsigned int)((double)topFactors.size() * retrain);
```

Comment:
Revert to `unsigned int` and cast the whole next line to `unsigned int` also.  Same with the update below.